diff --git a/src/getters.sh.c b/src/getters.sh.c
index 5e24ab3..086ae76 100644
--- a/src/getters.sh.c
+++ b/src/getters.sh.c
@@ -74,3 +74,17 @@ get_metafile() {
 	fi
 }
 
+get_use_desc() {
+	local use="$1"
+	local FILE
+	(
+		if [[ -e _SHAREDIR/pkg++/uses ]]; then
+			grep "$use" _SHAREDIR/pkg++/uses
+		fi
+		for FILE in _SYSCONFDIR/pkg++/\*.use; do
+			grep "$use" $FILE
+		done
+	) | tail -n 1 \
+	  | cut -d "|" -f 2
+}
+
