--- pkg++-0.5.99/pkg++.in	2010-05-06 12:23:55.422302693 +0200
+++ pkg++/pkg++.in	2010-05-06 12:36:55.929123406 +0200
@@ -552,7 +552,11 @@ build_package() {
 		
 		cd $PKG
 		info "Build result:"
-		tar czvvf $TARGET *
+		if [[ "$PKGMK_NEW_PACKAGES" = "yes" ]]; then
+			tar --xz -cvvf $TARGET *
+		else
+			tar czvvf $TARGET *
+		fi
 		
 		if [ $? = 0 ]; then
 			BUILD_SUCCESSFUL="yes"
@@ -702,6 +706,7 @@ print_help() {
 	echo "  -kw,  --keep-work           keep temporary working directory"
 	echo "  -cf,  --config-file <file>  use alternative configuration file"
 	echo "  -p,   --pkgfile <file>      use a pkgfile other than \`$PKGMK_PKGFILE'"
+	echo "  -np   --new-packages        make xz compressed packages (experimental)"
 	echo "  -v,   --version             print version and exit "
 	echo "  -h,   --help                print help and exit"
 }
@@ -768,6 +773,8 @@ parse_options() {
 				PKGMK_SHA256SUM="$DIR/.sha256sum"
 				PKGMK_NOSTRIP="$DIR/.nostrip"
 				shift;;
+			-np|--new-packages)
+				PKGMK_NEW_PACKAGES="yes" ;;
 			-v|--version)
 				echo "`basename $PKGMK_COMMAND` (pkgutils) $PKGMK_VERSION"
 				exit 0 ;;
@@ -815,7 +822,11 @@ main() {
 	
 	check_pkgfile
 	
-	TARGET="$PKGMK_PACKAGE_DIR/$name#$version-$release.pkg.tar.gz"
+	if [[ "$PKGMK_NEW_PACKAGES" = "yes" ]]; then
+		TARGET="$PKGMK_PACKAGE_DIR/$name#$version-$release.pkg.tar.xz"
+	else
+		TARGET="$PKGMK_PACKAGE_DIR/$name#$version-$release.pkg.tar.gz"
+	fi
 	
 	if [ "$PKGMK_CLEAN" = "yes" ]; then
 		clean
@@ -925,4 +936,6 @@ PKGMK_KERNEL="no-kernel"
 
 main "$@"
 
+PKGMK_NEW_PACKAGES="no"
+
 # End of file
